src/catalog/database_catalog.cpp

```diff
@@ -2193,6 +2198,12 @@ void DatabaseCatalog::BootstrapProcContexts(const common::ManagedPointer<transac
   SetProcCtxPtr(txn, postgres::TRIM2_PRO_OID, func_context);
   txn->RegisterAbortAction([=]() { delete func_context; });
 
+  func_context = new execution::functions::FunctionContext("concat", type::TypeId::VARCHAR,
+                                                           {type::TypeId::VARCHAR, type::TypeId::VARCHAR},
+                                                           execution::ast::Builtin::Concat, true);
+  SetProcCtxPtr(txn, postgres::CONCAT_PRO_OID, func_context);
+  txn->RegisterAbortAction([=]() { delete func_context; });
```

Comment:
I'm surprised we do so much registering of abort actions to cleanup pointers when other parts of catalog bootstrapping asserts that we can't fail. Do we envision this bootstrap txn ever failing, @tanujnay112?

Reply:
> I'm surprised we do so much registering of abort actions to cleanup pointers when other parts of catalog bootstrapping asserts that we can't fail. Do we envision this bootstrap txn ever failing, @tanujnay112?

I'm not really sure why all the abort actions are there. I tested it and it does work fine without them. Would the DB still even be able to start up if this txn aborted?

Reply:
At that moment, I just put this in because every other Bootstrapping function was doing so as well. But yeah now that I think about it we shouldn't have the bootstrapping transaction fail. Otherwise, the whole system pretty much does not proceed from there right.

Reply:
If that's the case I can remove this line from the PR, and I can open a separate PR to remove it from the other functions